target_chains/cosmwasm/tools/src/chains-manager/chain-querier.ts

```diff
@@ -0,0 +1,162 @@
+import { Tendermint34Client } from "@cosmjs/tendermint-rpc";
+import { QueryClient } from "@cosmjs/stargate";
+import { WasmExtension, setupWasmExtension } from "@cosmjs/cosmwasm-stargate";
+
+// TODO: expose the querier and consume them in price pusher
+
+/**
+ * Interface for classes implementing a querier for a cosmwasm chain.
+ *
+ * The querier interacts with contracts only and can get it's info, stored state.
+ * It can also query the `Query methods` defined in the contract.
+ *
+ * For contract dependent response, you need to look into contract specific schema in order
+ * to know what to expect
+ *
+ * @interface ChainQuerier
+ */
+export interface ChainQuerier {
+  /**
+   * `getContractInfo` gets the contract info for the give contract address.
+   * @param {ContractInfoRequest} req
+   * @param {string} req.contractAddr
+   * @returns {ContractInfoResponse}
+   * - {number} res.codeId. The codeId of the contract's code.
+   * - {string} res.creator. The creator of the contract.
+   * - {string} res.adminAddr - Address of the current admin.
+   * - {string} res.label - The label with which the contract was instantiated
+   *
+   * @throws an error if it fails.
+   */
+  getContractInfo(req: ContractInfoRequest): Promise<ContractInfoResponse>;
+
+  /**
+   * `getSmartContractState` query the `Query methods` implemented in the contract
+   * @param {SmartContractRequest} req
+   * @param {string} req.contractAddr
+   * @param {Object} req.query - The query object for the contract. It accepts any object as it is contract dependent.
+   * @returns {Object} - It returns an object. As the response is contract dependent we can't have a structure for it.
+   *
+   * @throws an error if it fails.
+   */
+  getSmartContractState(req: SmartContractRequest): Promise<Object>;
+
+  /**
+   * Contracts local storage on chain is structured as key-value pairs.
+   * `getSmartContractState` query the local storage of a contract for the given key.
+   * @param {RawContractStateRequest} req
+   * @param {string} req.contractAddr
+   * @param {Buffer} req.key
+   * @returns {Object} - It returns an object. As the response is contract dependent we can't have a structure for it.
+   *
+   * @throws an error if it fails.
+   */
+  getRawContractState(req: RawContractStateRequest): Promise<Object>;
+
+  /**
+   * Contracts local storage on chain is structured as key-value pairs.
+   * `getAllContractState` query the local storage of a contract for all such pairs.
+   * @param {AllContractStateRequest} req
+   * @param {string} req.contractAddr
+   * @returns {Object} - It returns an object. As the response is contract dependent we can't have a structure for it.
+   *
+   * @throws an error if it fails.
```

Comment:
this `@throws` text isn't very helpful -- it doesn't tell me the conditions under which it would fail, or how I should handle errors. I would either omit it entirely (if the error cases aren't particularly important and shouldn't happen), or describe the errors that can happen in a little more detail. If all of the methods on this interface can fail for roughly the same conditions (e.g., your network connection is out), you can document the error cases once in the class docstring and omit it on the functions.